kong/clustering/compat/init.lua

```diff
@@ -353,4 +356,56 @@ function _M.update_compatible_payload(config_table, dp_version, log_suffix)
 end
 
 
+-- returns has_update, modified_deflated_payload, err
+function _M.update_compatible_payload(payload, dp_version, log_suffix)
+  local cp_version_num = version_num(KONG_VERSION)
+  local dp_version_num = version_num(dp_version)
+
+  -- if the CP and DP have the same version, avoid the payload
+  -- copy and compatibility updates
+  if cp_version_num == dp_version_num then
+    return false
+  end
+
+  local has_update
+  payload = deep_copy(payload, false)
```

Comment:
Seems we are deep-copying unconditionally. Could we also include the lazy-eval optimization, just like what we did to wrpc version of this function?